### PR TITLE
Chore: Upgrade ESLint

### PR DESCRIPTION
**Pull request by tassoevan**
## Proposed changes (including videos or screenshots)

Upgrade ESLint (to 8.19.0) and its dependencies, dropping outdated rules.

### CODE REVIEW

**Review by d-gubert — DISMISSED** [review]
All good on Apps-Engine side

File: apps/meteor/ee/client/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1 @@
+../../client/.eslintrc.json
```
Comment (sampaiodiego) [inline apps/meteor/ee/client/.eslintrc.json:1]: can we stop using symbolic links? they're way harder to keep, someone needs to remember they exists and need to be taken care of.

we could simply use `"extends": "../../client/.eslintrc.json"`

File: apps/meteor/client/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,257 @@
+{
+  "root": true,
```
Comment (sampaiodiego) [inline apps/meteor/client/.eslintrc.json:2]: why do we still need a eslint config specific for client? why not merge it with the root one?

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

File: apps/meteor/client/.eslintrc.json (hunk 2 above)
Comment (tassoevan) [reply apps/meteor/client/.eslintrc.json:2]: My memory might be betraying me, but I think at the time some `eslint-react-plugin` rules conflicted with other ESLint rules (maybe due to ESLint version being outdated), as well as many of the configuration comes from the Fuselage monorepo. I wanted to review everything, but I thought this is a little bit out of the scope of this pull request.

**Review by sampaiodiego — APPROVED** [review]

**Review by sampaiodiego — APPROVED** [review]

**Review by ggazzo — APPROVED** [review]

**Pull request merged**